### PR TITLE
updated to Purescript 0.14

### PR DESCRIPTION
**Pull request by CarstenKoenig**
Hi,

as mentioned here: https://github.com/menelaos/purescript-stringutils/pull/7 I'm trying to help you out a bit with updating to Purescript 0.14

As a temporary fix (till stringutils is in the package-set) I included my fork here so that this will compile for now.

~~Sadly for this one I had some issues - for some reason `fromRight` does not really work as expected - I included a quick workaround `getRight` and will try to look into this more~~

Update: fixed thanks to @newlandsvalley

### CODE REVIEW

**Comment by newlandsvalley** [comment]
Probably need to use fromRight' - see https://github.com/purescript/documentation/blob/master/migration-guides/0.14-Migration-Guide.md#the-dataeitherfromleft-and-dataeitherfromright-functions-are-now-total-functions

**Comment by CarstenKoenig** [comment]
yeah - my bad (stupid me) - I fix this tomorrow

I actually saw that fromLeft/fromRight are total now, but I totally missed the *default* value there (really stupid sorry)

---

I guess I should better install the tooling first next time - after this I did and things worked out a lot better

**Comment by newlandsvalley** [comment]
Yeah, I missed it first time as well.

**Comment by CarstenKoenig** [comment]
ok - thanks for the hint - I think this is much better

**Comment by realvictorprm** [comment]
What a coincidence @CarstenKoenig meeting you here :smile:!
We're using that libary in our projects and I just wanted to update it to purescript v0.14.0, looks like you were faster so many thanks for updating :smiley:

File: test/Test/Data/String/Base64.purs [hunk 1]
```diff
@@ -65,14 +66,14 @@ testBase64 = do
...
 
   log "decode <<< encodeUrl == id for well-formed input"
   let
     encodeUrlDecodeIdProp :: WellFormedInput -> Result
     encodeUrlDecodeIdProp (WellFormedInput str) =
-      str === unsafePartial (fromRight <<< decode <<< encodeUrl $ str)
+      str === fromRight "err" (decode <<< encodeUrl $ str)
 
-  quickCheck encodeUrlDecodeIdProp
+  quickCheck encodeUrlDecodeIdProp
```
Comment (realvictorprm) [inline test/Test/Data/String/Base64.purs:79]: Newline is missing here :see_no_evil:

Comment (CarstenKoenig) [reply test/Test/Data/String/Base64.purs:79]: > What a coincidence @CarstenKoenig meeting you here !
> We're using that libary in our projects and I just wanted to update it to purescript v0.14.0, looks like you were faster so 

it's a small world - especially the FP one I guess

**Comment by CarstenKoenig** [comment]
seems a recent update made all the necessary changes so this obsolete

**Pull request closed without merging**